### DISCUSSION TITLE
why geopandas not support osm.pbf data format?

### DISCUSSION THREAD

**Discussion by daizzhisheng** [geopandas/geopandas#3015] [Q&A] [answered]
why geopandas not support osm.pbf data format?

someone said read file use 'pyogrio' engine，by ‘’gdal‘’ can read  osm.pbf， but it can't work.

**Comment by martinfleis** [comment, marked as answer]
"can't work" is a bit vague description of the error. 

GeoPandas depends on GDAL to read files. GDAL is exposed either through Fiona or Pyogrio. Whether either of them can read osm.pbf depends on whether the driver for this file format is included in GDAL and if it is enabled. Fiona generally does not include OSM driver but pyogrio does.

You may be hitting [this](https://github.com/geopandas/pyogrio/issues/272) issue, for which there is a fix coming in the next release of pyogrio or a workaround. So, with `pyogrio` installed using conda from conda-forge, this works:

```py
df = geopandas.read_file(
    "http://download.geofabrik.de/europe/andorra-latest.osm.pbf",
    engine="pyogrio", 
    use_arrow=True,
)
```

Without `use_arrow=True`, this returns `FeatureError: GDAL returned more records than expected based on the count of records that may meet your combination of filters against this dataset.` (which was fixed but not yet released). To allow `use_arrow=True`, you also need ´pyarrow` installed. But with that, it will work.

**Reply by daizzhisheng** [reply]
thanks for you beautiful reply!

**Reply by daizzhisheng** [reply]
someone said PYROMS is also a good solution, which can read osm.pbf file to geodataFrame.

**Reply by martinfleis** [reply]
pyrosm can do that but I would bet this is faster. pyrosm was also really memory hungry last time I tried.

**Reply by theroggy** [reply]
FYI: a new version of pyogrio has been released recently, so it should be solved now.
https://github.com/geopandas/pyogrio/releases
